codec/src/lib.rs

```diff
@@ -9,7 +9,7 @@
 //! of this code; in this way the client and server can more gracefully
 //! manage unknown enum variants.
 #![allow(dead_code)]
-#![cfg_attr(feature = "cargo-clippy", allow(clippy::range_plus_one))]
+#![cfg_attr(clippy, allow(clippy::range_plus_one))]
```

Comment:
From the docs, it sounds like all the cfg_attr stuff can be removed?

```suggestion
#![allow(clippy::range_plus_one)]
```